dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCachePositionReader.java

```diff
@@ -138,6 +139,28 @@ public synchronized void close() throws IOException {
     mFallbackReader.close();
   }
 
+  /**
+   * Get a {@link DataFileChannel} which wraps a {@link io.netty.channel.FileRegion}.
+   * @param position the start position to read
+   * @param length how many bytes to read
+   * @return an object of {@link DataFileChannel}
+   */
+  public DataFileChannel getDataFileChannel(long position, int length) {
+    long currentPage = position / mPageSize;
+    PageId pageId;
+    if (mCacheContext.getCacheIdentifier() != null) {
+      pageId = new PageId(mCacheContext.getCacheIdentifier(), currentPage);
+    } else {
+      pageId = new PageId(mFileId.toString(), currentPage);
+    }
+    int currentPageOffset = (int) (position % mPageSize);
+    int bytesLeftInPage = (int) (mPageSize - currentPageOffset);
+    int bytesToReadInPage = Math.min(bytesLeftInPage, length);
+    // If the page is not in Alluxio, we will return null here
+    return mCacheManager.getDataFileChannel(
+        pageId, currentPageOffset, bytesToReadInPage, mCacheContext);
```

Comment:
It's better to throw `PageNotFoundException` in case the page does not exist

Reply:
Used `Optional` instread.